lib/puppet/provider/package/zypper.rb

```diff
@@ -12,6 +12,30 @@
   defaultfor :operatingsystem => [:suse, :sles, :sled, :opensuse]
   confine    :operatingsystem => [:suse, :sles, :sled, :opensuse]
 
+  def self.latest_package_version(zypper_instance, package)
+    if @latest_versions.nil?
+      @latest_versions = list_updates(zypper_instance)
+    end
+
+    @latest_versions[package]
+  end
+
+  def self.list_updates(zypper_instance)
+    output = zypper_instance.zypper 'list-updates'
```

Comment:
Why is this passing `zypper_instance` and calling `zypper_instance.zypper` instead of just invoking `zypper`?